crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -347,7 +350,15 @@ impl<T: Send + Sync + Clone, W: Send + Sync> OnchainOrderParser<T, W> {
             self.settlement_contract,
             self.metrics,
         )
-        .await;
+        .await
+        .into_iter()
+        .filter(|(_, _, _, order)| {
+            if let Some(market_orders_deprecation_date) = self.market_orders_deprecation_date {
```

Comment:
Is this code path also hit for ethflow orders or does that need a separate change?

Reply:
Removed this code as I implemented it as Felix suggested.